lib/IDE/CodeCompletionCache.cpp

```diff
@@ -165,8 +166,8 @@ static bool readCachedModule(llvm::MemoryBuffer *in,
     }
 
     const char *p = strings + index;
-    auto size = read32le(p);
-    auto str = copyString(*V.Allocator, StringRef(p, size));
+    size_t size = read32le(p);
+    auto str = NullTerminatedStringRef(StringRef(p, size), *V.Allocator);
```

Comment:
This is used for `CodeCompletionString` too, so their chunk text may needlessly be null-terminated. But there's a high chance of that the chunk text is used as a filter name. So I don't think this increases memory usage much in total.

Reply:
I'd be more worried about the extra memory from the wrapped StringRef than the single byte from the null termination 🤷

Reply:
I thought memory footprint of `StringRef` object and `NullTerminatedStringRef` object is the same. Just wrapping it increase the memory?

Reply:
Sorry, you're correct 👍